### PR TITLE
Binder badge

### LINKED ISSUE #1107: add action to post binder link on each PR
we want to create a github action that posts a link to binder whenever someone opens a PR, this will allow us to easily test things

some notes:

1. since this requires a github token, I believe the trigger should be pull_request_target
2. the host should be binder.ploomber.io
3. we also need to add an environment.yml, similar to [this one](https://github.com/ploomber/jupysql/blob/master/environment.yml)
4. and a [postBuild](https://github.com/ploomber/jupysql/blob/master/postBuild) like this

more information

https://mybinder.readthedocs.io/en/latest/
https://mybinder.org/

some related code:
https://github.com/jupyterlab/maintainer-tools/blob/main/.github/actions/binder-link/action.yml

### PR DESCRIPTION
**Pull request by AnirudhVIyer**
## Describe your changes
Created a binder-badge wotkflow, environment.yml, and a postbuild
## Issue number

Closes #1107 

## Checklist before requesting a review

- [x] Performed a self-review of my code
- [x] Formatted my code with [`pkgmt format`](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#linting-formatting)
- [x] Added [tests](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#testing) (when necessary).
- [x] Added [docstring](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#documenting-changes-and-new-features) documentation and update the [changelog](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#changelog) (when needed)

----
:books: Documentation preview :books:: https://ploomber--1112.org.readthedocs.build/en/1112/

### CODE REVIEW

**Comment by edublancas** [comment]
@AnirudhVIyer: i see the tests are failing:

```
AttributeError: module 'mistune' has no attribute 'Renderer'
```

it might be due to the new mistune release: https://pypi.org/project/mistune/#history

**Comment by AnirudhVIyer** [comment]
> @AnirudhVIyer: i see the tests are failing:
> 
> ```
> AttributeError: module 'mistune' has no attribute 'Renderer'
> ```
> 
> it might be due to the new mistune release: https://pypi.org/project/mistune/#history

Yeah, I am checking that out, also i have some dependencies issues when setting up the ploomber env, trying to solve that, maybe some issue with that as well.

**Comment by AnirudhVIyer** [comment]
@edublancas are there any permissions needed to change the workflow files? Because i can't see the draft PR triggering the worflow file I added. Could you give me a pointer if there is something wrong with my `binder-badge.yml` file?

File: src/ploomber/util/markup.py [hunk 1]
```diff
@@ -23,15 +23,15 @@ def block_code(self, code, lang=None):
         return markdown(md)
     else:
 
-        class HighlightRenderer(mistune.Renderer):
+        class HighlightRenderer(mistune.HTMLRenderer):
```
Comment (edublancas) [inline src/ploomber/util/markup.py:26]: remove these changes and merge from master, I've already fixed this: https://github.com/ploomber/ploomber/commit/8acdf9a10eb9e207f14061c283a7450053ba8833

then I can merge and see if that triggers the action

**Review by edublancas — CHANGES_REQUESTED** [review]

File: src/ploomber/util/markup.py (hunk 1 above)
Comment (AnirudhVIyer) [reply src/ploomber/util/markup.py]: Sure!

**Pull request merged**